server/middleware/setUpAuthentication.ts

```diff
@@ -49,5 +50,12 @@ export default function setUpAuth(): Router {
     next()
   })
 
+  router.use(
+    rateLimiter({
+      windowMs: config.rateLimits.authentication.windowTime,
+      maxRequests: config.rateLimits.authentication.maxRequests,
+    }),
+  )
```

Comment:
this will still have the same problem as before, the router is not attached to any path and there is no isolation between routers - so by adding the router here you'll be adding it to every request.

look at the following for example: 
```
const express = require('express')
const app = express()

const router = express.Router()
app.use(router)

router.use((req, res, next) => {
  res.locals.msg = 'router 1'
  next()
})

app.get('/', (req, res) => {  res.send('Hello from: ' + res.locals.msg)})

app.listen(3000, () => { console.log(`Example app listening`) })
```
Hitting the endpoint will return `Hello from: router 1`

Also order really matters, adding middleware later will mean its visited later.
So in this case the rate limiter only kicks in after passport handles any requests.

Reply:
So, given that the only paths we have here are the sign in / out etc paths, is it fair to say they're the only ones we can/should ratelimit? Otherwise we're ratelimiting the entire stack?

Reply:
yes, I think so